core/src/main/scala/org/apache/spark/streaming/SnappyStreamingContext.scala

```diff
@@ -56,7 +59,31 @@ class SnappyStreamingContext protected[spark](
         "both SparkContext and checkpoint as null")
   }
 
-  val snappySession = new SnappySession(sc)
+  val snappySession = reuseSnappySession.getOrElse(new SnappySession(sc))
+  currentSnappySession.foreach(csn => {
```

Comment:
If reuseSnappySession is None, then won't currentSnappySession be used? Or that case should never arise?

Reply:
Yes that case will never arise, because the function from which currentSnappySession is passed, the reuseSnappySession is passed as none.